sim-lib/src/lib.rs

```diff
@@ -275,13 +276,50 @@ impl Simulation {
         Ok(())
     }
 
+    // validate_node_network validates that the nodes that are on the same bitcoin network.
+    // e.g. node A and B are on testnet, node C is on mainnet should return an error.
```

Comment:
Comment nitpick: it validates that the nodes are all on the same network and ensures that we're not running on mainnet.